src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -272,6 +269,102 @@ export class CollapseCellsAction extends ToggleableAction {
 	}
 }
 
+// Run Notebook with Parameters
+export class RunParametersAction extends TooltipFromLabelAction {
+	private static readonly label = localize('runParameters', "Run with Parameters");
+	private static readonly iconClass = 'icon-run-with-parameters';
+
+	constructor(id: string, toggleTooltip: boolean,
+		context: URI,
+		@IQuickInputService private quickInputService: IQuickInputService,
+		@INotebookService private _notebookService: INotebookService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(id, {
+			label: RunParametersAction.label,
+			baseClass: baseIconClass,
+			iconClass: RunParametersAction.iconClass,
+			maskedIconClass: maskedIconClass,
+			shouldToggleTooltip: toggleTooltip
+		});
+	}
+
+	/**
+	 * Gets Default Parameters in Notebook from Parameter Cell
+	 * Uses that as Placeholder values for user to inject new values for
+	 * Once user enters all values it will open the new parameterized notebook
+	 * with injected parameters value from the QuickInput
+	*/
+	public async run(context: URI): Promise<void> {
+		// Get Parameters via Notebook Model
+		const editor = this._notebookService.findNotebookEditor(context);
+		// Set Default Parameters to the mapped string
+		let defaultParameters = new Map<string, string>();
+		editor.cells.forEach(cell => {
+			if (cell.isParameter) {
+				for (let parameter of cell.source) {
+					let param = parameter.trim().split('=');
+					defaultParameters.set(param[0].trim(), param[1].trim());
+				}
+			}
+		});
+
+		// New Parameters to be set to injected parameters cell
+		let pick = new Map<string, string>();
+		let newParams: string = '';
+		let addParam: string;
+		let index = 0;
+		// Store new parameter values to map based off defaultParameters
+		if (defaultParameters.size === 0) {
+			// If there is no parameter cell indicate to user to create one
+			this.notificationService.notify({
+				severity: Severity.Info,
+				message: localize('noParametersCell', "Unable to Run with Parameters as there is no Parameter Cell. Create a Parameter Cell first. Learn more [here](https://docs.microsoft.com/en-us/sql/azure-data-studio/notebooks/notebooks-parameterization?view=sql-server-ver15)."),
+			});
+		} else {
+			for (let key of defaultParameters.keys()) {
+				let newParameterValue = await this.quickInputService.input({ prompt: key, placeHolder: defaultParameters.get(key) });
+				pick.set(key, newParameterValue);
```

Comment:
If the user cancels out of the quickpick this is going to be undefined - do you want to set that to the value at that point? (vs just leaving it as the original value)

Reply:
A user can click, or skip the first default parameter for example but enter the second parameter. If we keep it undefined then it will allow us to understand that the user only wants the second default parameter value to be updated and injected. (This is what the if condition within the next for loop checks)